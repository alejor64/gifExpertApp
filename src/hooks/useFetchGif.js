import { useState, useEffect } from "react"
import { getGift } from "../helpers/getGif"

export const useFetchGif = (category) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGift(category)
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                })
            })
    }, [category])

    return state
}