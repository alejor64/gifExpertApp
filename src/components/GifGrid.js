import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { useFetchGif } from '../hooks/useFetchGif'
import GifGridItem from './GifGridItem'

const GifGrid = ({category}) => {

    const {data: images, loading} = useFetchGif(category)

    return (
        <Fragment>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {
                loading && <p className="animate__animated animate__flash">Loading...</p>
            }
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem key={img.id} {...img} />
                        ))
                    }
            </div>
        </Fragment>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid
