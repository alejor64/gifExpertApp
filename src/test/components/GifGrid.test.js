import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import GifGrid from '../../components/GifGrid'
import { useFetchGif } from '../../hooks/useFetchGif'

jest.mock('../../hooks/useFetchGif')

describe('Prueba en el componente <GifGrid />', () => {
    const category = 'Dragon Ball'
    
    test('Debe renderizar el compoenente', () => {
        useFetchGif.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(true)
    })
    
    test('Debe de mostrar los item cuando se cargan imgs useFetchGif', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhosto/img.jpg',
            title: 'Cualquier nombre'
        }]

        useFetchGif.mockReturnValue({
            data: gifs,
            loading: false
        })
    
        const wrapper = shallow(<GifGrid category={category} />)
        
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
    
})
