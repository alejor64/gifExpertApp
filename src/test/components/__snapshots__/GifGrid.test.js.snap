// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente <GifGrid /> Debe de mostrar los item cuando se cargan imgs useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier nombre"
      url="https://localhosto/img.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba en el componente <GifGrid /> Debe renderizar el compoenente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
